meta-resin-common/classes/image_types_resin.bbclass

```diff
@@ -169,7 +169,7 @@ IMAGE_CMD_resin-sdcard () {
 	# Create a vfat filesystem with boot files
 	BOOT_BLOCKS=$(LC_ALL=C parted -s ${RESIN_SDIMG} unit b print | awk '/ 1 / { print substr($4, 1, length($4 -1)) / 512 /2 }')
 	rm -rf ${WORKDIR}/boot.img
-	mkfs.vfat -n "${RESIN_BOOT_FS_LABEL}" -S 512 -C ${WORKDIR}/boot.img $BOOT_BLOCKS
+	mkfs.vfat -n "${RESIN_BOOT_FS_LABEL}" -F 32 -S 512 -C ${WORKDIR}/boot.img $BOOT_BLOCKS
```

Comment:
I think Page is right here. mkfs.vfat will auto-select the appropriate FAT type based on the size you're trying to create.

If as you say the boot partition was bigger than whatever the threshold for 32bit FAT is, then mkfs.vfat would have selected it and MBR would have been consistent. Since we've ended up with partitions in MBR having inconsistent types it means that the boot partition is not more than the threshold, and forcing 32bit FAT seems like a bad idea.

Why not setting the MBR to be whatever mkfs.vfat autoselects?